### ISSUE TITLE
'Version' object has no attribute 'release'  in CircleCI docs deploy

### ISSUE THREAD

**Issue by jfalkenstein** [Sceptre/sceptre#1174]
We're encountering an issue in our CircleCI builds:

```
# github.com:22 SSH-2.0-babeld-e37038b1
Cloning into 'sceptre.github.io'...
Warning: Permanently added the RSA host key for IP address '140.82.114.4' to the list of known hosts.
remote: Enumerating objects: 4545, done.        
remote: Counting objects: 100% (2261/2261), done.        
remote: Compressing objects: 100% (1066/1066), done.        
remote: Total 4545 (delta 1821), reused 1602 (delta 1170), pack-reused 2284        
Receiving objects: 100% (4545/4545), 8.38 MiB | 41.45 MiB/s, done.
Resolving deltas: 100% (3542/3542), done.
Building docs in /home/circleci/docs/sceptre.github.io/dev
Traceback (most recent call last):
  File "/home/circleci/project/venv/bin/sphinx-apidoc", line 5, in <module>
    from sphinx.ext.apidoc import main
  File "/home/circleci/project/venv/lib/python3.8/site-packages/sphinx/ext/apidoc.py", line 30, in <module>
    from sphinx.cmd.quickstart import EXTENSIONS
  File "/home/circleci/project/venv/lib/python3.8/site-packages/sphinx/cmd/quickstart.py", line 40, in <module>
    from sphinx.util.template import SphinxRenderer
  File "/home/circleci/project/venv/lib/python3.8/site-packages/sphinx/util/template.py", line 22, in <module>
    from sphinx.jinja2glue import SphinxFileSystemLoader
  File "/home/circleci/project/venv/lib/python3.8/site-packages/sphinx/jinja2glue.py", line 20, in <module>
    from sphinx.application import TemplateBridge
  File "/home/circleci/project/venv/lib/python3.8/site-packages/sphinx/application.py", line 34, in <module>
    from sphinx.domains import Domain, Index
  File "/home/circleci/project/venv/lib/python3.8/site-packages/sphinx/domains/__init__.py", line 24, in <module>
    from sphinx.roles import XRefRole
  File "/home/circleci/project/venv/lib/python3.8/site-packages/sphinx/roles.py", line 20, in <module>
    from sphinx.util.docutils import ReferenceRole, SphinxRole
  File "/home/circleci/project/venv/lib/python3.8/site-packages/sphinx/util/docutils.py", line 44, in <module>
    __version_info__ = version.parse(docutils.__version__).release
AttributeError: 'Version' object has no attribute 'release'

Exited with code exit status 1
CircleCI received exit code 1
```

### Your environment
* Our CircleCI builds

**Comment by jfalkenstein** [comment]
@zaro0508 This issue seems directly related to the upgrade from `packaging` <17 to >17: 
![image](https://user-images.githubusercontent.com/77296393/143898534-e9199e46-9ba9-46d6-9ae4-ce1c79de2eea.png)

According to the [changelog](https://github.com/pypa/packaging/blob/main/CHANGELOG.rst#170---2017-02-28), the newer version of packaging has a `release` attribute. Our current requirements pin <17 for that package, but It appears as if other packages are now requiring later versions.

**Comment by jfalkenstein** [comment]
It appears as if a commit for Sphinx was recently added that made the breaking change for old versions of packaging: https://github.com/sphinx-doc/sphinx/commit/ee1cae8c1c2c0e3df9cd2343c32c83ad6142aaca#diff-8a06a6233264b57f905c460f3a647e5e55246f2e6ba9f0ed4deb87adb68c85d1

**Comment by jfalkenstein** [comment]
The root of this problem seems to be the recent release of Sphinx 4.3.1 combined with our _very_ old version of `packaging`. This combination blows up when sphinx imports [this line](https://github.com/sphinx-doc/sphinx/blob/4.x/sphinx/util/docutils.py#L44).

The short-term solution to this appears to be pinning our version of Sphinx < 4.3.1. However, that's not sustainable. What we _really_ need to do is Upgrade some of our antiquated package versions, like setuptools and packaging, both of which are set to versions from several _years_ ago.

**Comment by jfalkenstein** [comment]
Actually, I was wrong. This change was released with Sphinx 4.3.0, not 4.3.1. Short term fix in #1171 is to pin Sphinx at <4.3
